libtenzir/builtins/contexts/geoip.cpp

```diff
@@ -490,18 +510,19 @@ class ctx final : public virtual context {
   }
 
   auto save() const -> caf::expected<save_result> override {
-    auto builder = flatbuffers::FlatBufferBuilder{};
-    auto path = builder.CreateString(db_path_);
-    fbs::context::geoip::GeoIPDataBuilder geoip_builder(builder);
-    geoip_builder.add_url(path);
-    auto geoip_data = geoip_builder.Finish();
-    fbs::context::geoip::FinishGeoIPDataBuffer(builder, geoip_data);
-    return save_result{.data = tenzir::fbs::release(builder), .version = 1};
+    if (!mmdb_) {
+      return caf::make_error(ec::lookup_error,
+                             fmt::format("no GeoIP data currently exists for "
+                                         "this context"));
+    }
+    return save_result{.data = std::move(chunk::mmap(db_path_).value()),
+                       .version = latest_version};
   }
 
 private:
   std::string db_path_;
```

Comment:
We can remove the file path here and replace it with a memory-mapped chunk of the file contents for use in `save`.